commit 2c4c3500a92d2f74a474a3504c538a83c3acd598 (refs/changes/19/919/1)
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2016-11-17T17:08:43-08:00 (2 years, 11 months ago)
    
    DOCKER-977 sdc-docker failing to initialize buckets due to differences in how node-fast returns errors

diff --git a/lib/moray.js b/lib/moray.js
index 3b7e3f3..bcfaade 100644
--- a/lib/moray.js
+++ b/lib/moray.js
@@ -16,6 +16,8 @@ var assert = require('assert-plus');
 var async = require('async');
 var restify = require('restify');
 var util = require('util');
+var verror = require('verror');
+var VError = verror.VError;
 var vasync = require('vasync');
 
 
@@ -153,7 +155,7 @@ function initBucket(moray, bucket, callback) {
 
     moray.getBucket(bucket.name, function (err, oldBucket) {
         if (err) {
-            if (err.name === 'BucketNotFoundError') {
+            if (VError.hasCauseWithName(err, 'BucketNotFoundError')) {
                 moray.log.info(bucket.schema, 'initBucket: creating bucket %s',
                     bucket.name);
                 return moray.createBucket(bucket.name, bucket.schema,
@@ -259,7 +261,7 @@ function reindexBucket(moray, bucket, callback) {
  */
 function delObj(moray, bucket, key, callback) {
     moray.delObject(bucket.name, key, function (err) {
-        if (err && err.name === 'ObjectNotFoundError') {
+        if (err && VError.hasCauseWithName(err, 'ObjectNotFoundError')) {
             return callback(new restify.ResourceNotFoundError(err,
                 '%s not found', bucket.desc));
         }
@@ -280,7 +282,7 @@ function delObj(moray, bucket, key, callback) {
 function getObj(moray, bucket, key, callback) {
     moray.getObject(bucket.name, key, function (err, res) {
         if (err) {
-            if (err.name === 'ObjectNotFoundError') {
+            if (VError.hasCauseWithName(err, 'ObjectNotFoundError')) {
                 return callback(new restify.ResourceNotFoundError(err,
                     '%s not found', bucket.desc));
             }
diff --git a/package.json b/package.json
index 4e1b72b..cb137e8 100644
--- a/package.json
+++ b/package.json
@@ -29,7 +29,7 @@
     "triton-tags": "1.0.0",
     "ufds": "1.2.0",
     "vasync": "1.6.3",
-    "verror": "1.6.0",
+    "verror": "1.9.0",
     "wf-client": "0.2.0",
     "xregexp": "3.1.0"
   },
